### ISSUE TITLE
Don't rewrite a file when downloading an attachment with the same name

### ISSUE THREAD

**Issue by DenBond7** [FlowCrypt/flowcrypt-android#370]
Need to think about a situation when a user has some file `file.txt` in the Downloads and tries to download an attachment with the name `file.txt`. Currently, we rewrite such files. But the files can be absolutely different. Need to resolve such situations.

**Comment by tomholub** [comment]
you can add a counter, eg:
filename: `file.txt`
if already exists, try file.1.txt
if elready exits, try file.2.txt
(remember, if the file has no extension, you should put the number at the end, eg `file` -> `file.1`)

**Comment by DenBond7** [comment]
I'm closing it. It was fixed by https://github.com/FlowCrypt/flowcrypt-android/commit/9ea5752b7478a8ea2b0dc8b028ac88aefad26241

Just with one difference. Up to Android 9.0(include), we should look after the file name, after Android 10(include) the system decides what a name use. Android System uses the following pattern for existed files `filename (index).txt`. I decided to use the same logic for Android 9.0 and less.

Examples
- `file.txt`
- `file(1).txt`
- `file(2).txt`
- `file`
- `file(1)`
- `file(2)`
